extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -0,0 +1,490 @@
+			const tenants: Tenant[] = tenantResponse.data.value.map((tenantInfo: TenantResponse) => {
+				return {
+					id: tenantInfo.tenantId,
+					displayName: tenantInfo.displayName ? tenantInfo.displayName : localize('azureWorkAccountDisplayName', "Work or school account"),
+					userId: token.key,
+					tenantCategory: tenantInfo.tenantCategory
+				} as Tenant;
+			});
+
+			const homeTenantIndex = tenants.findIndex(tenant => tenant.tenantCategory === 'Home');
+			if (homeTenantIndex >= 0) {
+				const homeTenant = tenants.splice(homeTenantIndex, 1);
+				tenants.unshift(homeTenant[0]);
+			}
+
+			return tenants;
+		} catch (ex) {
+			console.log(ex);
+			throw new Error('Error retreiving tenant information');
+		}
+	}
+
+	protected async getSubscriptions(account: AzureAccount): Promise<Subscription[]> {
+		interface SubscriptionResponse { // https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list
+			subscriptionId: string
+			tenantId: string
+			displayName: string
+		}
+		const allSubs: Subscription[] = [];
+		const tokens = await this.getSecurityToken(account, azdata.AzureResource.ResourceManagement);
+
+		for (const tenant of account.properties.tenants) {
+			const token = tokens[tenant.id];
+			const subscriptionUri = url.resolve(this.metadata.settings.armResource.endpoint, 'subscriptions?api-version=2019-11-01');
+			try {
+				const subscriptionResponse = await this.makeGetRequest(token.token, subscriptionUri);
+				const subscriptions: Subscription[] = subscriptionResponse.data.value.map((subscriptionInfo: SubscriptionResponse) => {
+					return {
+						id: subscriptionInfo.subscriptionId,
+						displayName: subscriptionInfo.displayName,
+						tenantId: subscriptionInfo.tenantId
+					} as Subscription;
+				});
+				allSubs.push(...subscriptions);
+			} catch (ex) {
+				console.log(ex);
+				throw new Error('Error retreiving subscription information');
+			}
+		}
+		return allSubs;
+	}
+
+	protected async getToken(postData: { [key: string]: string }, tenant = this.commonTenant, resourceId: string = ''): Promise<TokenRefreshResponse | undefined> {
+		try {
+			const tokenUrl = `${this.loginEndpointUrl}${tenant}/oauth2/token`;
+
+			const tokenResponse = await this.makePostRequest(tokenUrl, postData);
+			const tokenClaims = this.getTokenClaims(tokenResponse.data.access_token);
+
+			const accessToken: AccessToken = {
+				token: tokenResponse.data.access_token,
+				key: tokenClaims.email || tokenClaims.unique_name || tokenClaims.name,
+			};
+
+			this.memdb.set(`${accessToken.key}_${tenant}_${resourceId}`, tokenResponse.data.expires_on);
+
+			const refreshToken: RefreshToken = {
+				token: tokenResponse.data.refresh_token,
+				key: accessToken.key
+			};
+
+			return { accessToken, refreshToken, tokenClaims };
+
+		} catch (err) {
+			console.dir(err);
+			const msg = localize('azure.noToken', "Retrieving the token failed.");
+			throw new Error(msg);
```

Comment:
I think you missed showing the localized error to the user here instead of throwing the localized error.